File: BaekJoon Online Judge/C++/Step08-Math_1 Mathematical Thinking/1193.cpp
//
//  1193.cpp
//  Step08-Math_1 Mathematical Thinking
//
//  Created by Yebin Kim on 2019/10/23.
//  Copyright © 2019 김예빈. All rights reserved.
//

#include <iostream>

using namespace std;

// Time Complexity O(n*log(n))
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, a = 1, up, down;
    
    cin >> n;
    
    while(n > 0) n -= a++;
    
    up = n + --a;
    down = ++a - up;
    
    if (a % 2 != 0)
        cout << up << "/" << down;
    else
        cout << down << "/" << up;
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step08-Math_1 Mathematical Thinking/1193.swift
//
//  1193.swift
//  Step08-Math_1 Mathematical Thinking
//
//  Created by Yebin Kim on 2019/11/13.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

if var N = Int(readLine()!) {
    var num = 1
    var up = 0
    var down = 0
    
    while N > 0 {
        N -= num
        num += 1
    }
    
    num -= 1
    up = N + num
    
    num += 1
    down = num - up
    
    if num % 2 == 0 {
        print("\(down)/\(up)")
    } else  {
        print("\(up)/\(down)")
    }
}
